recipes/default.rb

```diff
@@ -10,26 +10,14 @@
   compile_time false if respond_to?(:compile_time)
 end
 
-if platform_family? 'rhel'
-  yum_repository 'influxdb' do
-    description 'InfluxDB Repository - RHEL \$releasever'
-    baseurl node['influxdb']['upstream_repository']
-    gpgkey 'https://repos.influxdata.com/influxdb.key'
+influxdb_install 'influxdb' do
+  if node['kernel']['machine'].include? '64'
+    arch_type 'amd64'
+  elsif node['kernel']['machine'].include? '386'
+    arch_type 'i386'
+  else
+    Chef::Log.warning "I am unable to determine arch type for: #{node['kernel']['machine']}"
   end
```

Comment:
I thought about it while implementing and was torn as I do like the idea of having someone having a specified option in the provider in the case of improper detection. I was going for max flexibility in the provider. Let me know if you would either like this removed from the recipe, moved into the provider, or left as is.